doc/stdlib/dates.rst

```diff
@@ -535,6 +535,31 @@ Periods
 
    Construct a ``Period`` type with the given ``v`` value. Input must be losslessly convertible to an ``Int64``\ .
 
+.. function:: CompoundPeriod(periods) -> CompoundPeriod
+
+   .. Docstring generated from Julia source
+
+   Construct a ``CompoundPeriod`` from a ``Vector`` of ``Period``\ s. The provided ``Period``\ s will be simplified using the following rules:
+
+   * All ``Period``\ s of the same type will be added together
```

Comment:
It's important to document what `CompoundPeriod` is *for*.  Maybe something like:

> `CompoundPeriod` is useful for expression time periods that are not a fixed multiple of smaller periods.  For example, "a year and a day" is not a fixed number of days, but can be expressed using `CompoundPeriod`.   In fact, a `CompoundPeriod` is automatically generated by addition of incommensurate periods, e.g. `Dates.Year(1) + Dates.Day(1)` produces a `CompoundPeriod` result. 
> When a `CompoundPeriod` is constructed, it is automatically simplified into a canonical form according to the following rules: